### DISCUSSION TITLE
PDF Freeze When Loading

### DISCUSSION THREAD

**Discussion by kennygokh** [RyotaUshio/obsidian-pdf-plus#222] [Q&A]
Hi, I noticed that when I open a PDF, it will freeze for a moment. After a few seconds then it started to allow to scroll. It happen when you click a link within a note for a PDF. Regardless of whether it is open within the same panel or another panel, same behavior. Any idea what went wrong? Any settings that I could check?

**Comment by RyotaUshio** [comment]
The next version will include some performance improvement around the loading process although I'm not sure how much it will be helpful in the exact situation you're in. I will re-notify you when it's available.

**Reply by RyotaUshio** [reply]
I've released 0.40.8: https://github.com/RyotaUshio/obsidian-pdf-plus/releases/tag/0.40.8

If you don't need the citation hover popover feature (see the image below), setting the ["Hover(+Ctrl/Cmd) on a citation link to show..."](obsidian://pdf-plus?setting=actionOnCitationHover) option to "Same as other internal links" will improve the performance. Even if you do, I've done some optimization around this feature so you might (or might not) notice some improvement.

<img width="508" alt="image" src="https://github.com/RyotaUshio/obsidian-pdf-plus/assets/72342591/3713f976-6ace-4304-8d73-709396703337">

**Reply by kennygokh** [reply]
Performance slightly improve.

**Reply by kennygokh** [reply]
Hi @RyotaUshio, I noticed that the slowness issue came back. I have already disabled the option that you mentioned. Any other options that I need to do? To be more exact, the issue is more apparent when copying text from PDF and to paste into .md file. I am using 0.40.8.
